apps/server/src/modules/system/controller/system.controller.ts

```diff
@@ -1,20 +1,24 @@
 import { Controller, Get, Query } from '@nestjs/common';
 import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
-import { SystemOauthResponse } from '@src/modules/system/controller/dto/system-oauth.response';
 import { SystemFilterParams } from '@src/modules/system/controller/dto/system.filter.params';
-import { SystemOauthResponseMapper } from '@src/modules/system/controller/mapper/system-oauth-response.mapper';
 import { SystemUc } from '../uc/system.uc';
+import { PublicSystemListResponse } from './dto/public-system-list.response';
+import { SystemResponseMapper } from './mapper/system-response.mapper';
 
-@ApiTags('System')
-@Controller('system')
+@ApiTags('Systems')
+@Controller('systems')
 export class SystemController {
 	constructor(private readonly systemUc: SystemUc) {}
 
-	@Get()
-	@ApiOperation({ summary: 'Finds all systems.' })
-	@ApiResponse({ status: 200, type: SystemOauthResponse, description: 'Returns an SystemOauthResponse.' })
-	async find(@Query() filterParams: SystemFilterParams): Promise<SystemOauthResponse> {
+	/**
+	 * This endpoint is used to show users the possible login systems that exist.
+	 * No sensible data should be returned!
+	 */
+	@Get('public')
+	@ApiOperation({ summary: 'Finds all publicly available systems.' })
+	@ApiResponse({ status: 200, type: PublicSystemListResponse, description: 'Returns a list of systems.' })
+	async find(@Query() filterParams: SystemFilterParams): Promise<PublicSystemListResponse> {
```

Comment:
are there cases for fetching systems other then oauth? Where is that used? I found only the legacy client reference
It is unclear to me why are we allowing for example to fetch all the systems, like iServ or ldap, or ldap univention in the db and returning their ids...
Also, for ldap for example, one must check also that ldapConfig.active is true 
Not sure if there is some extra config for other systems types

Reply:
To address your points:
Currently there are only use cases for OAuth Systems, however we have planned to use this endpoint as well for at least LDAP systems. 
Therefore, it should stay generic imo.

This Endpoint replaces the old one that fetched all data for the login list and features the additional use for the migration page. 
Both pages need the same info and we were fine before as well.

There currently are no use cases for logged in users to fetch all systems. 
Even splitting it for load balancing would still result in the same amount of DB queries in the end.

Good point about the queries. We have a Story in progress (N21-504) that will enable Redis in NestJS which can be used natively to cache API Responses.
We can use that to enable caching here (and all other endpoints which might need it!) and solve this concern.

About the LDAP Systems: good point, we will keep it in mind for when we handle this story. It is currently Out of Scope though.

Reply:
reference for the ldapConfig.active thing: https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/src/services/school/services/schoolsList.js#L17

Reply:
the one security concern is that this enpoint will imediatly and public expose ids of all systems, including instance wide  systems like iserv, tsp, ldap univention, and also any system (ldap) which is actually not active